src/sys/windows/stdnet/addr.rs

```diff
@@ -0,0 +1,170 @@
+use std::ascii;
+use std::fmt;
+use std::io;
+use std::mem;
+use std::os::raw::c_int;
+use std::path::Path;
+
+use windows_sys::Win32::Networking::WinSock::{sockaddr_un, SOCKADDR};
+
+fn path_offset(addr: &sockaddr_un) -> usize {
+    // Work with an actual instance of the type since using a null pointer is UB
+    let base = addr as *const _ as usize;
+    let path = &addr.sun_path as *const _ as usize;
+    path - base
+}
+
+cfg_os_poll! {
+    use windows_sys::Win32::Networking::WinSock::AF_UNIX;
+    pub(super) fn socket_addr(path: &Path) -> io::Result<(sockaddr_un, c_int)> {
```

Comment:
I've reviewed way to many of this function... up to 4 now...

Reply:
sorry 😬 will aim to not make cosmetic changes to functions mid-review going forward